comp/core/workloadmeta/workloadmeta.go

```diff
@@ -138,12 +142,70 @@ func newWorkloadMetaOptional(deps dependencies) optionalProvider {
 	c := newWorkloadMeta(deps)
 
 	return optionalProvider{
-		Comp:          optional.NewOption(c.Comp),
-		FlareProvider: c.FlareProvider,
-		Endpoint:      c.Endpoint,
+		Comp:                         optional.NewOption(c.Comp),
+		FlareProvider:                c.FlareProvider,
+		WorkloadListEndpoint:         c.WorkloadListEndpoint,
+		PodContainerMetadataEndpoint: c.PodContainerMetadataEndpoint,
 	}
 }
 
+type PodContainerMetadata struct {
+	Name       string                `json:"name"`
+	Cmd        []string              `json:"cmd"`
+	Entrypoint []string              `json:"entrypoint"`
+	Image      OrchestratorContainer `json:"image"`
+}
+
+func (wm *workloadmeta) podContainerMetadata(name, ns string) (map[string]PodContainerMetadata, error) {
+	if ns == "" || name == "" {
+		return nil, fmt.Errorf("missing pod name or namespaces")
+	}
+
+	pod, err := wm.GetKubernetesPodByName(name, ns)
+	if err != nil {
+		return nil, err
+	}
+
+	out := map[string]PodContainerMetadata{}
+	for _, c := range pod.Containers {
+		image, err := wm.GetImage(c.Image.ImageMetadataID())
+		if err != nil {
+			return out, fmt.Errorf("could not get image for container %s: %w", c.Name, err)
+		}
+
+		out[c.Name] = PodContainerMetadata{
+			Name: c.Name,
+			Image: c,
```

Comment:
same question, here `c` is a container metadata of type `OrchestratorContainer`. Why are we assigning it to a field called `Image`. 

Probably either field type or field name should be fixed.